examples/llm/src/models/layers/attention.py

```diff
@@ -347,6 +379,129 @@ def forward(self,
         return self.out_proj(context), attn_weights, past_key_value
 
 
+class MultiQueryAttention(nn.Module):
+    """Multi-Query self attention.
+
+    Using torch or triton attention implemetation enables user to also use
+    additive bias.
+    """
+
+    def __init__(
+        self,
+        d_model: int,
+        n_heads: int,
+        attn_impl: str = 'triton',
+        attn_clip_qkv: Optional[float] = None,
+        attn_qk_ln: bool = False,
+        softmax_scale: Optional[float] = None,
+        attn_pdrop: float = 0.0,
+        low_precision_layernorm: bool = False,
```

Comment:
this probably gets refactored after your norm PR goes in? Just heads up

Reply:
yeah merge conflicts make this one pretty clear...